packages/modul-components/src/components/svg/svg.ts

```diff
@@ -0,0 +1,133 @@
+import { PluginObject } from 'vue';
+import Component from 'vue-class-component';
+import { Emit, Prop } from 'vue-property-decorator';
+import { eraseTag, eraseTagAndAllIsContent } from '../../utils/clean/htmlClean';
+import { Enums } from '../../utils/enums/enums';
+import { ModulIconName } from '../../utils/modul-icons/modul-icons';
+import { REGEX_CSS_NUMBER_VALUE } from '../../utils/props-validation/props-validation';
+import { ModulVue } from '../../utils/vue/vue';
+import { SVG_NAME } from '../component-names';
+import WithRender from './svg.html';
+
+@WithRender
+@Component
+export class MSvg extends ModulVue {
+    @Prop({
+        default: ModulIconName.Error,
+        validator: value => Enums.toValueArray(ModulIconName).includes(value)
+    })
+    public readonly name: ModulIconName;
+
+    @Prop({ default: '' })
+    public readonly customSvg: string;
+
+    @Prop({
+        default: '1em',
+        validator: (value: string) =>
+            REGEX_CSS_NUMBER_VALUE.test(value)
+    })
+    public readonly width: string;
+
+    @Prop({
+        default: '1em',
+        validator: (value: string) =>
+            REGEX_CSS_NUMBER_VALUE.test(value)
+    })
+    public readonly height: string;
+
+    @Prop()
+    public readonly svgTitle: string;
+
+    @Emit('click')
+    public emitClick(event: MouseEvent): void { }
+
+    @Emit('keydown')
+    public emitKeydown(event: KeyboardEvent): void { }
+
+    @Emit('mouseover')
+    public emitMouseOver(event: MouseEvent): void { }
+
+    @Emit('mouseleave')
+    public emitMouseLeave(event: MouseEvent): void { }
+
+    @Emit('svg-id')
+    public emitSvgId(svgId: string): void { }
+
+    public async onDataSvgIdChange(svgId: string | undefined): Promise<void> {
+        if (!svgId) {
+            return;
+        }
+        await this.$nextTick();
+        this.emitSvgId(svgId);
+    }
+
+    public get svg(): string {
+        if (this.customSvg) {
+            return this.customSvg;
+        }
+
+        if (!this.name) {
+            return '';
+        }
+
+        try {
+            return require(`./../../assets/icons/svg/${this.name}.svg`);
```

Comment:
on ne peut pas faire de wildcard dans un required , ca cause des problème au niveau de webpack qui va inclure touts les svg du repertoire /icons/svg dans le bundle et on se ramasse alors avec les 200 svg dans le javascript

ca va être a enlever ici et dans message-page.ts

On s'appelle pour en discuter!

on s'appel pour en discuter mais

Reply:
Je viens de retirer le require et j'ai créé un SvgSpriteService (svg-sprite.ts) où nous pouvons ajouter les icônes à la pièce (voir https://github.com/ulaval/modul/issues/225).

Reply:
Je crois qu'il y a plusieurs améliorations avec le SvgSpriteService, mais c'est un bon début

Reply:
Super! 

Il faudra regler le probleme avec le message-page.ts. Étant donner qu'il fait un     👎 

```
protected created(): void {
        if (this.svgName) {
            this.svg = require(`../../assets/icons/svg/${this.svgName}.svg`);
        }
    }
```

Cela importe tout les svgs que tu as mit dans le package assets/icons/svg/. Avant c'était pas une probleme parce qu'il avait qu'un dizaine de svg. Maitenant que tu en as mit 200 ca les load tous.

2 solutions posiible

1. modifier le message-page pour qu'il importe seulement les svg qui on rapport a ce composant et non un ${this.svgName}.svg (dans cette solution message-page doit tenir une genre de map en memoire avec ses svgs

2. Changer message-page pour quil utilise un autre repetoire que  assets/icons/svg/ et deplacer dans ce repertoire les svg qui on rapport avec ce composant (ca pourrait etre /src/component/message-page/svg

Reply:
La PR est maintenant prête.